adagrams/game.py

```diff
@@ -1,13 +1,126 @@
 from random import randint
 
 def draw_letters():
-    pass
+    tiles = [
+    'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 
+    'B', 'B',
+    'C', 'C',
+    'D', 'D', 'D', 'D', 
+    'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 
+    'F', 'F', 
+    'G', 'G', 'G', 
+    'H', 'H',  
+    'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 
+    'J',  
+    'K', 
+    'L', 'L', 'L', 'L', 
+    'M', 'M', 
+    'N', 'N', 'N', 'N', 'N', 'N', 
+    'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 
+    'P', 'P', 
+    'Q', 
+    'R', 'R', 'R', 'R', 'R', 'R', 
+    'S', 'S', 'S', 'S', 
+    'T', 'T', 'T', 'T', 'T', 'T',
+    'U', 'U', 'U', 'U', 
+    'V', 'V', 
+    'W', 'W', 
+    'X', 
+    'Y', 'Y', 
+    'Z'
+    ]
+    selected_index = []
+    picked_letters = []
+    for i in range(0, 10):
+        rand_selection = randint(0, len(tiles)-1)
+        while rand_selection in selected_index:
+            rand_selection = randint(0, len(tiles)-1)
+        selected_index.append(rand_selection)
+        random_letter = tiles[rand_selection]
+        picked_letters.append(random_letter)
+    return picked_letters
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word_uppercase = word.upper()
+    letter_frequency_word = {}
+    letter_frequency_hand = {}
+    
+    for i in word_uppercase:
+        if i not in letter_bank:
+            return False
+        else:
+            letter_frequency_word[i] = letter_frequency_word.get(i, 0) + 1
+
+    for i in letter_bank:
+        letter_frequency_hand[i] = letter_frequency_hand.get(i, 0) + 1
+        
+    for letter in word_uppercase:
+        if letter_frequency_word[letter] > letter_frequency_hand[letter]:
+            return False
+
+    return True
 
 def score_word(word):
-    pass
+
+    word_uppercase = word.upper()
+    point_sum = 0
+
+    #point lists
+    one_point_letters = ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T" ]
+    two_point_letters = ["D", "G"]
+    three_point_letters = ["B", "C", "M", "P"]
+    four_point_letters = ["F", "H", "V", "W", "Y"]
+    five_point_letters = ["K"]
+    eight_point_letters = ["J", "X"]
+    ten_point_letters = ["Q", "Z"]
+
+    for letter in word_uppercase:
+        if letter in one_point_letters:
+            point_sum += 1
+        elif letter in two_point_letters:
+            point_sum += 2
+        elif letter in three_point_letters:
+            point_sum += 3
+        elif letter in four_point_letters:
+            point_sum +=4
+        elif letter in five_point_letters:
+            point_sum += 5
+        elif letter in eight_point_letters:
+            point_sum += 8
+        elif letter in ten_point_letters:
+            point_sum += 10
+
+    if len(word_uppercase) >= 7:
+        point_sum += 8
```

Comment:
8 is another magic number.

```suggestion
        BONUS_POINTS_FOR_LENGTH = 8
        point_sum += BONUS_POINTS_FOR_LENGTH
```